Decompiler/Decompiler.cs

```diff
@@ -788,6 +788,14 @@ private void DumpVPK(string parentPath, Package package, string type, Dictionary
             }
 
             var fileLoader = new BasicVpkFileLoader(package);
+            var gltfModelExporter = new GltfModelExporter
```

Comment:
You can specify multiple threads in decompiler, and gltf exporter isnt thread safe, so it's very likely that this would cause issues. Keep it as it currently is.

Reply:
This specific function - DumpVPK - is not affected by threads setting, and furthermore it shouldn't be, as it'd race condition against itself on exporting same files.

In my tests of making `foreach (var file in entries)` multithreaded via queue (not committed anywhere) speeds it up from ~10m to ~6m for Dota 2 `-d -f models/ -e vmdl_c`, but the question of how to make it not race condition would still be open.

Reply:
> is not affected by threads setting
it is if you follow the call graph up to ParseVPK and then ProcessFile.

Exporter has DstDir which makes it not thread safe.

Reply:
> it is if you follow the call graph up to ParseVPK and then ProcessFile.

That path doesn't touch DumpVPK as far as I can tell.

Reply:
Anyway this can be resolved as `Don't reuse cache between glTF exports` makes glTF exporter thread-safe.

Reply:
Since you made the cache local, probably better to pass it as an argument, so at least that solves this particular thread safety.

> That path doesn't call DumpVPK as far as I can tell
I think it is.

> glTF exporter inside DumpVPK is not passed to any other threads whatsoever.
This you're right about, as it's local.

Reply:
> Since you made the cache local, probably better to pass it as an argument

It was already only applicable within a single exported file, and would break if, for example, one mesh was referenced in different files in a folder.
It only applied to meshes without skeleton before those changes though, which probably just hid the problem for the most part.

Anyway yes, now it is passed as an argument now to avoid try/catches.